Calculator/Calculator/LinkedList.swift

```diff
@@ -0,0 +1,146 @@
+//
+//  LinkedList.swift
+//  Calculator
+//
+//  Created by Whales on 2023/06/03.
+//
+
+class Node<T> {
+    var data: T?
+    var next: Node?
+    
+    init(data: T? = nil, next: Node? = nil) {
+        self.data = data
+        self.next = next
+    }
+}
+
+struct LinkedList<T> {
+    private var head: Node<T>?
+    private var tail: Node<T>?
+    private(set) var count: Int = 0
+    
+    var isEmpty: Bool {
+        head == nil
+    }
+    
+    func readHeadData() -> T? {
+        return head?.data
+    }
+    
+    func readTailData() -> T? {
+        return tail?.data
+    }
+    
+    mutating func append(data: T?) {
+        if isEmpty {
+            head = Node(data: data)
+            tail = head
+            count += 1
+            return
+        }
+        
+        let newNode = Node(data: data)
+        tail?.next = newNode
+        tail = newNode
+        count += 1
+    }
+    
+    mutating func insert(data: T?, at index: UInt) {
+        if index >= count {
+            return
+        } else if index == 0 {
```

Comment:
out of index의 상황을 고려한 로직이군요! 👍👍👍
고민하신 흔적이 보여서 좋습니다. 다만 로직의 특성상 guard를 쓰는게 더 좋아보여요. guard문은 if문과 어떤 차이가 있을까요?
```suggestion
        guard index < count else { return } 
        
        if index == 0 {
```

Reply:
`if`는 만약 `~면 ~해라`라는 의미로 쓰이는 반면에
`guard`는 `~가 아니면 끝내라`라는 의미로 쓰이면서 `early exit`이라는 특징을 가집니다.
그런 의미에서 로직의 특성상 `guard`를 쓰는게 더 좋아보인다고 말씀하신걸로 이해를 하고 코드를 수정하였습니다!!